### PR TITLE
Convert `aten.hardsigmoid` to `ttnn.hardsigmoid`

### LINKED ISSUE #480: aten.hardsigmoid.default

### PR DESCRIPTION
**Pull request by jdh8**
### Ticket
- Resolves #480

### Problem description
Compile `aten.hardsigmoid` to `ttnn.sigmoid`:
https://pytorch.org/docs/stable/generated/torch.nn.Sigmoid.html

### What's changed
- [x] Implement the conversion
- [x] Pass auto-generated op test cases
- [x] Pass testing models
- [x] Write op test cases

### CODE REVIEW

**Comment by jdh8** [comment]
Running tests with CI at:
https://github.com/tenstorrent/pytorch2.0_ttnn/actions/runs/11978897014

while I'm working on other issues

File: tests/lowering/eltwise/unary/test_hardsigmoid.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        (1, 16, 1, 1),
+        (1, 96, 1, 1),
+        (1, 240, 1, 1),
+        (1, 144, 1, 1),
+        (1, 288, 1, 1),
+        (1, 576, 1, 1),
+        (2, 3, 4, 5),
+        (7, 7, 7),
+        (420, 69),
+        pytest.param((1337,), marks=pytest.mark.xfail(reason="1D cases solved in #198, waiting for review")),
```
Comment (jdh8) [inline tests/lowering/eltwise/unary/test_hardsigmoid.py:38]: I added 4 custom cases for robustness.  1-D elementwise unary ops occasionally fail.  I solved this in #198.

Comment (ayerofieiev-tt) [reply tests/lowering/eltwise/unary/test_hardsigmoid.py:38]: Please fire a ticket so that I can report back to the team

**Review by ayerofieiev-tt — APPROVED** [review]

File: tests/lowering/eltwise/unary/test_hardsigmoid.py (hunk 1 above)
Comment (jdh8) [reply tests/lowering/eltwise/unary/test_hardsigmoid.py:38]: - tenstorrent/tt-metal#12671 ?

Comment (jdh8) [reply tests/lowering/eltwise/unary/test_hardsigmoid.py:38]: Oh no, the workaround in #198 stops working.  I'll find another way when I'm working on `aten.ceil`.

Comment (jdh8) [reply tests/lowering/eltwise/unary/test_hardsigmoid.py:38]: #198 is good again
